lib/logmsg/tests/test_tags.c

```diff
@@ -177,37 +166,68 @@ test_filters(gboolean not)
 
   for (i = 0; i < FILTER_TAGS; i++)
     {
-      test_msg("Testing filter, message has tag %d\n", i);
+      g_printf("Testing filter, message has tag %d\n", i);
       log_msg_set_tag_by_id(msg, i);
 
       if (((i % 3 == 1) ^ filter_expr_eval(f, msg)) ^ not)
-        test_fail("Failed to match message by tag %d\n", i);
+        g_printf("Failed to match message by tag %d\n", i);
 
-      test_msg("Testing filter, message no tag\n");
+      g_printf("Testing filter, message no tag\n");
       log_msg_clear_tag_by_id(msg, i);
       if (filter_expr_eval(f, msg) ^ not)
-        test_fail("Failed to match message with no tags\n");
+        g_printf("Failed to match message with no tags\n");
     }
 
   filter_expr_unref(f);
   log_msg_unref(msg);
 }
 
-int
-main(int argc, char *argv[])
+Test(tags, test_filters_false)
```

Comment:
This is a variation of the previous test based on a parameter.
Parameterized tests can be used here.

https://criterion.readthedocs.io/en/stable/parameterized.html
https://github.com/balabit/syslog-ng/blob/5a03be7255736941a55ed67fb27c6a08c84dd275/tests/unit/test_findcrlf.c